### PR TITLE
Fix 1.8 Istiod with 1.7 proxy with Sidecar.Ingress

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/30437

Put a hold for @Stono to verify

Note: This went directly to the 1.8 branch, as we do not want it on master or 1.9. Please review it as an original PR, not a cherrypick

### CODE REVIEW

**Comment by Stono** [comment]
Confirm i've tested this and it fixed https://github.com/istio/istio/issues/30437

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 1]
```diff
@@ -307,10 +307,20 @@ func (configgen *ConfigGeneratorImpl) buildInboundClusters(cb *ClusterBuilder, i
...
 
 		// For each workload port, we will construct a cluster
-		for _, instances := range clustersToBuild {
-			instance := instances[0]
-			localCluster := cb.buildInboundClusterForPortOrUDS(cb.proxy, int(instance.Endpoint.EndpointPort), actualLocalHost, instance, instances)
-			clusters = cp.conditionallyAppend(clusters, localCluster)
+		if !util.IsIstioVersionGE18(cb.proxy) {
+			// For 1.7 clusters, we used an old cluster format
+			for _, instances := range clustersToBuild {
+				for _, instance := range instances {
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/cluster.go:313]: I donot quite understand how this works? 

are there multi services selected the same pod?

Comment (howardjohn) [reply pilot/pkg/networking/core/v1alpha3/cluster.go:313]: yes, that is what this change is needed for

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 2]
```diff
@@ -352,7 +362,7 @@ func (configgen *ConfigGeneratorImpl) buildInboundClusters(cb *ClusterBuilder, i
 			instance.Endpoint.ServicePortName = listenPort.Name
 			instance.Endpoint.EndpointPort = uint32(port)
 
-			localCluster := cb.buildInboundClusterForPortOrUDS(nil, int(ingressListener.Port.Number), endpointAddress, instance, nil)
+			localCluster := cb.buildInboundClusterForPortOrUDS(cb.proxy, int(ingressListener.Port.Number), endpointAddress, instance, nil)
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/cluster.go:365]: I noticed this is the right fix, @Stono  you set sidecar ingress

Comment (howardjohn) [reply pilot/pkg/networking/core/v1alpha3/cluster.go:365]: Yes, there are 2 fixes - this part fixes the sidecar.ingress part and the other change fixes multiple services

**Comment by hzxuzhonghu** [comment]
lgtm

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by Stono** [comment]
Can we get this merged pretty please?  I'm keen to crack on with the 1.8 migration!

**Review by GregHanson — APPROVED** [review]

**Pull request merged**